typedargs/metadata.py

```diff
@@ -75,19 +75,79 @@ def __init__(self, func, name=None):
 
         self.load_from_doc = False
         self._doc_parsed = False
+        self._annotations_parsed = False
         self._docstring = docstring
 
+        # fixme: for some reason __init__ does not have __annotations__ here
+        self._type_annotations = getattr(func, '__annotations__', {})
+
     def _ensure_loaded(self):
-        if self.load_from_doc and not self._doc_parsed:
-            params, ret_info = parse_docstring(self._docstring)
-            for param_name, param_info in params.items():
-                self.add_param(param_name, param_info.type_name, param_info.validators)
 
-            if ret_info is not None:
-                self.return_info = ret_info
+        if self._annotations_parsed and self.load_from_doc and self._doc_parsed:
```

Comment:
This function looks like it has gotten too complicated.  For maintainability it should be refactored into several simpler functions that are composed to achieve the same objective.  Preferred options for the standalone functions are:

1. internal private module functions not in the class:

    ```python
    def _helper_function(arg1):
        """Do some internal simple thing in this module.
    ```

2. private methods inside the same class

The goal is to keep each function short and clear while minimizing the overall size of classes.  If an internal function can be written without referencing `self` or ` cls`, it's preferable to make it a standalone private function inside the module at the end of the module.

Reply:
I agree.. Moved a part of code to separate method.